### PR TITLE
feat: `tryParse()` now accepts `str?`

### PR DESCRIPTION
**Pull request by eladb**
The use case:

```js
let res = http.get("/foo");
let body = Json.tryParse(res.body);
```

Since `res.body` is `str?` (which makes sense), it also makes sense that `tryParse` just return `nil` when takes `nil`.

Previously, I had to do something like this:

```js
Json.tryParse(res.body ?? "");
```

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [x] Docs updated (only required for features)
- [x] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://www.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: libs/wingsdk/src/std/json.ts [hunk 1]
```diff
@@ -109,12 +109,12 @@ export class Json {
   /**
    * Try to parse a string into a Json
    *
-   * @macro ((args) => { try { return JSON.parse(args); } catch (err) { return undefined; } })($args$)
+   * @macro ((args) => { try { return args && JSON.parse(args); } catch (err) { return undefined; } })($args$)
```
Comment (tsuf239) [inline libs/wingsdk/src/std/json.ts:112]: if you'll put here `false`, it'll return `false` instead of `nil`, the same with `0`, I think

Comment (eladb) [reply libs/wingsdk/src/std/json.ts:112]: Since the signature is `str?` it won't be possible to pass these values from wing.

Comment (tsuf239) [reply libs/wingsdk/src/std/json.ts:112]: and what about passing `""` ? it'll return `""`

Comment (eladb) [reply libs/wingsdk/src/std/json.ts:112]: Nice catch. It should return `undefined` in that case. Checking.

Comment (eladb) [reply libs/wingsdk/src/std/json.ts:112]: Fixed.

**Review by MarkMcCulloh — APPROVED** [review]

**Pull request merged**